### PR TITLE
DEVEXP-484 Now supporting all aggregates

### PR DESCRIPTION
**Pull request by rjrudin**

### CODE REVIEW

File: src/main/java/com/marklogic/spark/reader/ReadContext.java [hunk 1]
```diff
@@ -161,23 +168,44 @@ void pushDownTopN(SortOrder[] orders, int limit) {
...
             .map(groupBy -> PlanUtil.expressionToColumnName(groupBy))
             .collect(Collectors.toList());
 
-        addOperatorToPlan(PlanUtil.buildGroupByCount(columnNames));
+        addOperatorToPlan(PlanUtil.buildGroupByAggregation(groupByColumnNames, aggregation));
+
+        StructType newSchema = buildSchemaWithColumnNames(groupByColumnNames);
+
+        for (AggregateFunc func : aggregation.aggregateExpressions()) {
+            if (func instanceof Avg) {
```
Comment (rjrudin) [inline src/main/java/com/marklogic/spark/reader/ReadContext.java:181]: This is ugly, I may need to refactor, the method is too long.

File: src/test/java/com/marklogic/spark/reader/PushDownGroupByAvgTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        verifyRows(
+            "avg(LuckyNumber)",
+            newDefaultReader()
+                .option(Options.READ_OPTIC_QUERY, QUERY_WITH_NO_QUALIFIER)
+                .load()
+                .groupBy("CitationID")
+                .agg(avg("LuckyNumber"))
+                .orderBy("CitationID")
+        );
+    }
```
Comment (BillFarber) [inline src/test/java/com/marklogic/spark/reader/PushDownGroupByAvgTest.java:39]: For the previous two tests: they produce the exact same output? What does .agg() do?

Comment (rjrudin) [reply src/test/java/com/marklogic/spark/reader/PushDownGroupByAvgTest.java:39]: I believe `agg()` is typically used to combine multiple aggregations, but it works with a single one. I added a test like that for each aggregate function to ensure that they behave the same way.

**Review by BillFarber — DISMISSED** [review]
Other than my question about the test

**Review by BillFarber — DISMISSED** [review]

**Review by BillFarber — DISMISSED** [review]
LTGM

**Review by BillFarber — DISMISSED** [review]

**Review by BillFarber — APPROVED** [review]

**Pull request merged**